File: web/schemas/admin/payments.py
from datetime import datetime
from typing import List, Optional

from pydantic import BaseModel


class AdminPaymentListItem(BaseModel):
    payment_id: int
    user_id: int
    username: Optional[str] = None
    first_name: Optional[str] = None
    amount: float
    original_amount: Optional[float] = None
    discount_applied: Optional[float] = None
    currency: str
    status: str
    provider: Optional[str] = None
    subscription_duration_months: Optional[int] = None
    promo_code: Optional[str] = None
    created_at: Optional[datetime] = None

    model_config = {"from_attributes": True}


class AdminPaymentListResponse(BaseModel):
    items: List[AdminPaymentListItem]
    total: int
    page: int
    page_size: int


class DailyRevenuePoint(BaseModel):
    date: str
    amount: float
    count: int


class ProviderRevenueItem(BaseModel):
    provider: str
    amount: float
    count: int


class PaymentStatsResponse(BaseModel):
    today_revenue: float
    week_revenue: float
    month_revenue: float
    all_time_revenue: float
    today_payments_count: int
    by_provider: List[ProviderRevenueItem]
    daily_chart: List[DailyRevenuePoint]

File: frontend/src/api/admin/payments.ts
import { apiRequest } from '@/api/client'

export interface AdminPaymentListItem {
  payment_id: number
  user_id: number
  username: string | null
  first_name: string | null
  amount: number
  original_amount: number | null
  discount_applied: number | null
  currency: string
  status: string
  provider: string | null
  subscription_duration_months: number | null
  promo_code: string | null
  created_at: string | null
}

export interface AdminPaymentListResponse {
  items: AdminPaymentListItem[]
  total: number
  page: number
  page_size: number
}

export interface DailyRevenuePoint {
  date: string
  amount: number
  count: number
}

export interface ProviderRevenueItem {
  provider: string
  amount: number
  count: number
}

export interface PaymentStatsResponse {
  today_revenue: number
  week_revenue: number
  month_revenue: number
  all_time_revenue: number
  today_payments_count: number
  by_provider: ProviderRevenueItem[]
  daily_chart: DailyRevenuePoint[]
}

export interface PaymentsListParams {
  status?: string
  provider?: string
  date_from?: string
  date_to?: string
  user_id?: number
  page?: number
  page_size?: number
}

export function getAdminPayments(params: PaymentsListParams = {}): Promise<AdminPaymentListResponse> {
  const sp = new URLSearchParams()
  if (params.status) sp.set('status', params.status)
  if (params.provider) sp.set('provider', params.provider)
  if (params.date_from) sp.set('date_from', params.date_from)
  if (params.date_to) sp.set('date_to', params.date_to)
  if (params.user_id !== undefined) sp.set('user_id', String(params.user_id))
  if (params.page !== undefined) sp.set('page', String(params.page))
  if (params.page_size !== undefined) sp.set('page_size', String(params.page_size))
  const qs = sp.toString()
  return apiRequest<AdminPaymentListResponse>(`/admin/payments${qs ? `?${qs}` : ''}`)
}

export function getPaymentStats(days = 30): Promise<PaymentStatsResponse> {
  return apiRequest<PaymentStatsResponse>(`/admin/payments/stats?days=${days}`)
}
